internal/mode/static/state/graph/graph.go

```diff
@@ -78,6 +83,17 @@ func (g *Graph) IsReferenced(resourceType client.Object, nsname types.Namespaced
 		_, existed := g.ReferencedNamespaces[nsname]
 		exists := isNamespaceReferenced(obj, g.Gateway)
 		return existed || exists
+	// Service reference exists if at least one HTTPRoute references it.
+	case *v1.Service:
+		_, exists := g.ReferencedServicesNames[nsname]
+		return exists
+	// EndpointSlice reference exists if its Service owner is referenced by at least one HTTPRoute.
+	case *discoveryV1.EndpointSlice:
```

Comment:
One of the things I'm unsure of here is since ReferencedServicesNames can have the NamespacedName of a Service that doesn't actually exist in the Cluster, should I change this logic such that it checks if its in the ReferencedServicesNames AND is in the cluster.Services? Is the scenario of an HTTPRoute pointing to a non-existing Service something that I should handle here? 

Another thing is: Can an EndpointSlice resource change its Service owner? If so then this won't work as `svcName` would no longer match any of the ReferencedServicesNames and thus will return false when it should return true.

Any other thoughts on this logic?